[Bug][功能]: EPD全分离场景，单机ipv6，QPS0.78,1.28，数据集image4，，一共发送200个请求，发送请求过程中，前100个请求请求正常，后100个请求出现Encode cache miss报错

### Your current environment

<details>

vllm commit link: tag v0.11.0rc4-EPD

</details>


### 🐛 Describe the bug

[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m INFO 12-17 12:21:18 [mooncake_store_connector_v1.py:220] Reqid: 18da939f-4f99-4fcc-a081-c5d8a00e84e3, Total tokens 3642, mooncake hit tokens: 384, need to load: 0
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m INFO 12-17 12:21:18 [mooncake_store_connector_v1.py:220] Reqid: 18da939f-4f99-4fcc-a081-c5d8a00e84e3, Total tokens 3642, mooncake hit tokens: 384, need to load: 0
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m INFO 12-17 12:21:18 [mooncake_store_connector_v1.py:220] Reqid: 18da939f-4f99-4fcc-a081-c5d8a00e84e3, Total tokens 3642, mooncake hit tokens: 384, need to load: 0
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m INFO 12-17 12:21:18 [mooncake_store_connector_v1.py:220] Reqid: 18da939f-4f99-4fcc-a081-c5d8a00e84e3, Total tokens 3642, mooncake hit tokens: 384, need to load: 0
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m INFO 12-17 12:21:18 [mooncake_store_connector_v1.py:220] Reqid: 18da939f-4f99-4fcc-a081-c5d8a00e84e3, Total tokens 3642, mooncake hit tokens: 384, need to load: 0
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m INFO 12-17 12:21:18 [mooncake_store_connector_v1.py:220] Reqid: 18da939f-4f99-4fcc-a081-c5d8a00e84e3, Total tokens 3642, mooncake hit tokens: 384, need to load: 0
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m INFO 12-17 12:21:18 [mooncake_store_connector_v1.py:220] Reqid: 18da939f-4f99-4fcc-a081-c5d8a00e84e3, Total tokens 3642, mooncake hit tokens: 384, need to load: 0
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m INFO 12-17 12:21:18 [mooncake_store_connector_v1.py:220] Reqid: 18da939f-4f99-4fcc-a081-c5d8a00e84e3, Total tokens 3642, mooncake hit tokens: 384, need to load: 0
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m INFO 12-17 12:21:18 [mooncake_store_connector_v1.py:220] Reqid: 18da939f-4f99-4fcc-a081-c5d8a00e84e3, Total tokens 3642, mooncake hit tokens: 384, need to load: 0
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m INFO 12-17 12:21:18 [mooncake_store_connector_v1.py:220] Reqid: 18da939f-4f99-4fcc-a081-c5d8a00e84e3, Total tokens 3642, mooncake hit tokens: 384, need to load: 0
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m INFO 12-17 12:21:18 [mooncake_store_connector_v1.py:220] Reqid: 18da939f-4f99-4fcc-a081-c5d8a00e84e3, Total tokens 3642, mooncake hit tokens: 384, need to load: 0
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m INFO 12-17 12:21:18 [mooncake_store_connector_v1.py:220] Reqid: 18da939f-4f99-4fcc-a081-c5d8a00e84e3, Total tokens 3642, mooncake hit tokens: 384, need to load: 0
[P_0] : I20251217 12:21:18.983324 281412288049568 pybind_client.cpp:1236] batch_put_from_multi_buffers: 1105688us
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m INFO 12-17 12:21:18 [mooncake_store_connector_v1.py:220] Reqid: 18da939f-4f99-4fcc-a081-c5d8a00e84e3, Total tokens 3642, mooncake hit tokens: 384, need to load: 0
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m INFO 12-17 12:21:19 [mooncake_store_connector_v1.py:220] Reqid: 18da939f-4f99-4fcc-a081-c5d8a00e84e3, Total tokens 3642, mooncake hit tokens: 384, need to load: 0
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m INFO 12-17 12:21:19 [mooncake_store_connector_v1.py:220] Reqid: 276253ed-58e8-46b1-a20a-77ee3d3ca98a, Total tokens 3642, mooncake hit tokens: 0, need to load: 0
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [mooncake_storage_connector.py:92] Load failed for 04aa9eda2f42c6ad6af7907bd40d4a9e4915cff8905f059efe8b0da3951a456e
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [mooncake_storage_connector.py:92] Load failed for 3a3e3e148b71796f4004f9dba627a78c27e080951122c092f4b8081830843d96
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [mooncake_storage_connector.py:92] Load failed for 7ea0341fbc2aa8486ac3af0dce42340df84b545199d82d65aca49d177cbfc926
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:69] Dumping input data for V1 LLM engine (v0.11.0) with config: model='/data/models/Qwen2.5-VL-7B-Instruct', speculative_config=None, tokenizer='/data/models/Qwen2.5-VL-7B-Instruct', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, tokenizer_revision=None, trust_remote_code=False, dtype=torch.bfloat16, max_seq_len=10000, download_dir=None, load_format=auto, tensor_parallel_size=1, pipeline_parallel_size=1, data_parallel_size=1, disable_custom_all_reduce=True, quantization=None, enforce_eager=True, kv_cache_dtype=auto, device_config=npu, structured_outputs_config=StructuredOutputsConfig(backend='auto', disable_fallback=False, disable_any_whitespace=False, disable_additional_properties=False, reasoning_parser=''), observability_config=ObservabilityConfig(show_hidden_metrics_for_version=None, otlp_traces_endpoint=None, collect_detailed_traces=None), seed=0, served_model_name=/data/models/Qwen2.5-VL-7B-Instruct, enable_prefix_caching=True, chunked_prefill_enabled=True, pooler_config=None, compilation_config={"level":0,"debug_dump_path":"","cache_dir":"","backend":"","custom_ops":["all"],"splitting_ops":null,"use_inductor":true,"compile_sizes":[],"inductor_compile_config":{"enable_auto_functionalized_v2":false},"inductor_passes":{},"cudagraph_mode":0,"use_cudagraph":true,"cudagraph_num_of_warmups":1,"cudagraph_capture_sizes":[],"cudagraph_copy_inputs":false,"full_cuda_graph":false,"use_inductor_graph_partition":false,"pass_config":{},"max_capture_size":0,"local_cache_dir":null}, 
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76] Dumping scheduler output for model execution: SchedulerOutput(scheduled_new_reqs=[NewRequestData(req_id=18da939f-4f99-4fcc-a081-c5d8a00e84e3,prompt_token_ids_len=3642,mm_features=[MultiModalFeatureSpec(data={'image_grid_thw': MultiModalFieldElem(modality='image', key='image_grid_thw', data=tensor([ 1, 40, 40]), field=MultiModalBatchedField()), 'pixel_values': MultiModalFieldElem(modality='image', key='pixel_values', data=tensor([[ 0.0618,  0.2520,  0.4258,  ...,  0.7539,  0.7812,  0.7969],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [ 1.8984,  1.9141,  1.9141,  ...,  0.6094,  0.5820,  0.5391],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [ 0.0618,  0.0908,  0.1201,  ..., -1.1406, -1.1953, -1.2500],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         ...,
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [ 1.7578,  1.8125,  1.8594,  ...,  0.9531,  0.8945,  0.8242],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [-0.0845, -0.0112,  0.0618,  ..., -0.7969, -0.8828, -0.9531],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [ 0.7031,  0.7188,  0.7344,  ..., -0.7539, -0.6562, -0.5430]],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]        dtype=torch.bfloat16), field=MultiModalFlatField(slices=[[slice(0, 1600, None)], [slice(1600, 3200, None)], [slice(3200, 4800, None)], [slice(4800, 6400, None)]], dim=0))}, modality='image', identifier='833e30e86edbe6990c2b63afee90052acd2925020ba803a3b1b5e1a03544d060', mm_position=PlaceholderRange(offset=15, length=400, is_embed=None)), MultiModalFeatureSpec(data={'image_grid_thw': MultiModalFieldElem(modality='image', key='image_grid_thw', data=tensor([ 1, 40, 40]), field=MultiModalBatchedField()), 'pixel_values': MultiModalFieldElem(modality='image', key='pixel_values', data=tensor([[1.7109, 1.7109, 1.7109,  ..., 1.9297, 1.9297, 1.9297],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [1.7109, 1.7109, 1.7109,  ..., 1.9297, 1.9297, 1.9297],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [1.7109, 1.7109, 1.7109,  ..., 1.9297, 1.9297, 1.9297],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         ...,
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [1.7109, 1.7109, 1.7109,  ..., 1.9297, 1.9297, 1.9297],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [1.7109, 1.7109, 1.7109,  ..., 1.9297, 1.9297, 1.9297],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [1.7109, 1.7109, 1.7109,  ..., 1.9297, 1.9297, 1.9297]],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]        dtype=torch.bfloat16), field=MultiModalFlatField(slices=[[slice(0, 1600, None)], [slice(1600, 3200, None)], [slice(3200, 4800, None)]], dim=0))}, modality='image', identifier='04aa9eda2f42c6ad6af7907bd40d4a9e4915cff8905f059efe8b0da3951a456e', mm_position=PlaceholderRange(offset=415, length=400, is_embed=None)), MultiModalFeatureSpec(data={'image_grid_thw': MultiModalFieldElem(modality='image', key='image_grid_thw', data=tensor([ 1, 40, 40]), field=MultiModalBatchedField()), 'pixel_values': MultiModalFieldElem(modality='image', key='pixel_values', data=tensor([[-1.7891, -1.7891, -1.7891,  ..., -1.4766, -1.4766, -1.4766],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [-1.7891, -1.7891, -1.7891,  ..., -1.4766, -1.4766, -1.4766],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [-1.7891, -1.7891, -1.7891,  ..., -1.4766, -1.4766, -1.4766],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         ...,
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [-1.7891, -1.7891, -1.7891,  ..., -1.4766, -1.4766, -1.4766],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [-1.7891, -1.7891, -1.7891,  ..., -1.4766, -1.4766, -1.4766],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [-1.7891, -1.7891, -1.7891,  ..., -1.4766, -1.4766, -1.4766]],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]        dtype=torch.bfloat16), field=MultiModalFlatField(slices=[[slice(0, 1600, None)], [slice(1600, 3200, None)], [slice(3200, 4800, None)]], dim=0))}, modality='image', identifier='3a3e3e148b71796f4004f9dba627a78c27e080951122c092f4b8081830843d96', mm_position=PlaceholderRange(offset=815, length=400, is_embed=None)), MultiModalFeatureSpec(data={'image_grid_thw': MultiModalFieldElem(modality='image', key='image_grid_thw', data=tensor([ 1, 40, 40]), field=MultiModalBatchedField()), 'pixel_values': MultiModalFieldElem(modality='image', key='pixel_values', data=tensor([[1.7109, 1.7109, 1.7109,  ..., 1.9297, 1.9297, 1.9297],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [1.7109, 1.7109, 1.7109,  ..., 1.9297, 1.9297, 1.9297],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [1.7109, 1.7109, 1.7109,  ..., 1.9297, 1.9297, 1.9297],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         ...,
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [1.7109, 1.7109, 1.7109,  ..., 1.9297, 1.9297, 1.9297],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [1.7109, 1.7109, 1.7109,  ..., 1.9297, 1.9297, 1.9297],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]         [1.7109, 1.7109, 1.7109,  ..., 1.9297, 1.9297, 1.9297]],
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:76]        dtype=torch.bfloat16), field=MultiModalFlatField(slices=[[slice(0, 1600, None)], [slice(1600, 3200, None)], [slice(3200, 4800, None)]], dim=0))}, modality='image', identifier='7ea0341fbc2aa8486ac3af0dce42340df84b545199d82d65aca49d177cbfc926', mm_position=PlaceholderRange(offset=1215, length=400, is_embed=None))],sampling_params=SamplingParams(n=1, presence_penalty=0.0, frequency_penalty=0.0, repetition_penalty=1.2, temperature=0.5, top_p=0.7, top_k=10, min_p=0.0, seed=77, stop=[], stop_token_ids=[], bad_words=[], include_stop_str_in_output=False, ignore_eos=True, max_tokens=1, min_tokens=0, logprobs=None, prompt_logprobs=None, skip_special_tokens=True, spaces_between_special_tokens=True, truncate_prompt_tokens=None, structured_outputs=None, extra_args=None),block_ids=([755, 756, 757, 471, 472, 473, 648, 649, 650, 651, 652, 653, 654, 655, 656, 657, 658, 659, 660, 661, 662, 663, 664, 665, 666, 1461, 494, 495, 496],),num_computed_tokens=384,lora_request=None,prompt_embeds_shape=None)], scheduled_cached_reqs=CachedRequestData(req_ids=[], resumed_from_preemption=[], new_token_ids=[], new_block_ids=[], num_computed_tokens=[]), num_scheduled_tokens={18da939f-4f99-4fcc-a081-c5d8a00e84e3: 3258}, total_num_scheduled_tokens=3258, scheduled_spec_decode_tokens={}, scheduled_encoder_inputs={}, num_common_prefix_blocks=[0], finished_req_ids=[], free_encoder_mm_hashes=['6875a81947294dae6362bab33aac45858e1966407c4fb5ea8752516df516e75b', 'eb71b219d583f72fa625a1aad8643ddb0f2bf4aef8540cdc6b6365990de8fadd', 'f46cbf2d3ca2bb5e509d346ab49a364640a62e6dff20ff96b57090c648756b5b', 'dea85871ea9f4af943de87b68c59fd005fe14a6e288a63c6f5764dd8faae6ef6'], structured_output_request_ids={}, grammar_bitmask=null, kv_connector_metadata=MooncakeConnectorMetadata(requests=[ReqMeta(req_id='18da939f-4f99-4fcc-a081-c5d8a00e84e3', token_ids=Tensor(shape=torch.Size([3584]), device=cpu,dtype=torch.int64), block_ids=[755, 756, 757, 471, 472, 473, 648, 649, 650, 651, 652, 653, 654, 655, 656, 657, 658, 659, 660, 661, 662, 663, 664, 665, 666, 1461, 494, 495, 496], save_spec=SaveSpec(skip_leading_tokens=0, can_save=true), load_spec=null, is_last_chunk=true, mm_features=[MultiModalFeatureSpec(data=MultiModalKwargsItem(data={pixel_values: MultiModalFieldElem(modality='image', key='pixel_values', data=Tensor(shape=torch.Size([1600, 1176]), device=cpu,dtype=torch.bfloat16), field=MultiModalFlatField(slices=[[slice(0, 1600, None)], [slice(1600, 3200, None)], [slice(3200, 4800, None)], [slice(4800, 6400, None)]], dim=0)), image_grid_thw: MultiModalFieldElem(modality='image', key='image_grid_thw', data=Tensor(shape=torch.Size([3]), device=cpu,dtype=torch.int64), field=MultiModalBatchedField())}, _modality='image'), modality='image', identifier='833e30e86edbe6990c2b63afee90052acd2925020ba803a3b1b5e1a03544d060', mm_position=PlaceholderRange(offset=15, length=400, is_embed=null)), MultiModalFeatureSpec(data=MultiModalKwargsItem(data={pixel_values: MultiModalFieldElem(modality='image', key='pixel_values', data=Tensor(shape=torch.Size([1600, 1176]), device=cpu,dtype=torch.bfloat16), field=MultiModalFlatField(slices=[[slice(0, 1600, None)], [slice(1600, 3200, None)], [slice(3200, 4800, None)]], dim=0)), image_grid_thw: MultiModalFieldElem(modality='image', key='image_grid_thw', data=Tensor(shape=torch.Size([3]), device=cpu,dtype=torch.int64), field=MultiModalBatchedField())}, _modality='image'), modality='image', identifier='04aa9eda2f42c6ad6af7907bd40d4a9e4915cff8905f059efe8b0da3951a456e', mm_position=PlaceholderRange(offset=415, length=400, is_embed=null)), MultiModalFeatureSpec(data=MultiModalKwargsItem(data={pixel_values: MultiModalFieldElem(modality='image', key='pixel_values', data=Tensor(shape=torch.Size([1600, 1176]), device=cpu,dtype=torch.bfloat16), field=MultiModalFlatField(slices=[[slice(0, 1600, None)], [slice(1600, 3200, None)], [slice(3200, 4800, None)]], dim=0)), image_grid_thw: MultiModalFieldElem(modality='image', key='image_grid_thw', data=Tensor(shape=torch.Size([3]), device=cpu,dtype=torch.int64), field=MultiModalBatchedField())}, _modality='image'), modality='image', identifier='3a3e3e148b71796f4004f9dba627a78c27e080951122c092f4b8081830843d96', mm_position=PlaceholderRange(offset=815, length=400, is_embed=null)), MultiModalFeatureSpec(data=MultiModalKwargsItem(data={pixel_values: MultiModalFieldElem(modality='image', key='pixel_values', data=Tensor(shape=torch.Size([1600, 1176]), device=cpu,dtype=torch.bfloat16), field=MultiModalFlatField(slices=[[slice(0, 1600, None)], [slice(1600, 3200, None)], [slice(3200, 4800, None)]], dim=0)), image_grid_thw: MultiModalFieldElem(modality='image', key='image_grid_thw', data=Tensor(shape=torch.Size([3]), device=cpu,dtype=torch.int64), field=MultiModalBatchedField())}, _modality='image'), modality='image', identifier='7ea0341fbc2aa8486ac3af0dce42340df84b545199d82d65aca49d177cbfc926', mm_position=PlaceholderRange(offset=1215, length=400, is_embed=null))])], unfinished_request_ids=['18da939f-4f99-4fcc-a081-c5d8a00e84e3']), ec_connector_metadata=ECMooncakeStorageConnectorMetadata(mm_datas=[MMMeta(mm_hash='04aa9eda2f42c6ad6af7907bd40d4a9e4915cff8905f059efe8b0da3951a456e', num_token=400), MMMeta(mm_hash='3a3e3e148b71796f4004f9dba627a78c27e080951122c092f4b8081830843d96', num_token=400), MMMeta(mm_hash='7ea0341fbc2aa8486ac3af0dce42340df84b545199d82d65aca49d177cbfc926', num_token=400)]))
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [dump_input.py:79] Dumping scheduler stats: SchedulerStats(num_running_reqs=1, num_waiting_reqs=68, step_counter=0, current_wave=0, kv_cache_usage=0.9959266802443992, prefix_cache_stats=PrefixCacheStats(reset=False, requests=2, queries=7284, hits=384), spec_decoding_stats=None, kv_connector_stats=None, num_corrupted_reqs=0)
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60] EngineCore encountered a fatal error.
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60] Traceback (most recent call last):
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]   File "/vllm-workspace/vllm-ascend/vllm_ascend/patch/platform/patch_core.py", line 51, in run_engine_core
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]     engine_core.run_busy_loop()
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]   File "/vllm-workspace/vllm/vllm/v1/engine/core.py", line 728, in run_busy_loop
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]     self._process_engine_step()
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]   File "/vllm-workspace/vllm/vllm/v1/engine/core.py", line 754, in _process_engine_step
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]     outputs, model_executed = self.step_fn()
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]                               ^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]   File "/vllm-workspace/vllm/vllm/v1/engine/core.py", line 284, in step
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]     model_output = self.execute_model_with_error_logging(
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]   File "/vllm-workspace/vllm/vllm/v1/engine/core.py", line 270, in execute_model_with_error_logging
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]     raise err
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]   File "/vllm-workspace/vllm/vllm/v1/engine/core.py", line 261, in execute_model_with_error_logging
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]     return model_fn(scheduler_output)
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]            ^^^^^^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]   File "/vllm-workspace/vllm/vllm/v1/executor/abstract.py", line 103, in execute_model
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]     output = self.collective_rpc("execute_model",
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]   File "/vllm-workspace/vllm/vllm/executor/uniproc_executor.py", line 83, in collective_rpc
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]     return [run_method(self.driver_worker, method, args, kwargs)]
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]   File "/vllm-workspace/vllm/vllm/utils/__init__.py", line 3122, in run_method
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]     return func(*args, **kwargs)
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]            ^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]   File "/vllm-workspace/vllm-ascend/vllm_ascend/worker/worker_v1.py", line 277, in execute_model
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]     output = self.model_runner.execute_model(scheduler_output,
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]   File "/usr/local/python3.11.13/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]     return func(*args, **kwargs)
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]            ^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]   File "/vllm-workspace/vllm-ascend/vllm_ascend/worker/model_runner_v1.py", line 1969, in execute_model
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]     max_query_len) = (self._prepare_inputs(scheduler_output,
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]   File "/vllm-workspace/vllm-ascend/vllm_ascend/worker/model_runner_v1.py", line 1407, in _prepare_inputs
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]     mm_embeds, is_mm_embed = self._gather_mm_embeddings(
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]   File "/vllm-workspace/vllm-ascend/vllm_ascend/worker/model_runner_v1.py", line 1099, in _gather_mm_embeddings
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]     assert encoder_output is not None, \
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60]            ^^^^^^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m ERROR 12-17 12:21:19 [patch_core.py:60] AssertionError: Encoder cache miss for 04aa9eda2f42c6ad6af7907bd40d4a9e4915cff8905f059efe8b0da3951a456e.
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m Process EngineCore_DP0:
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m Traceback (most recent call last):
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m   File "/usr/local/python3.11.13/lib/python3.11/multiprocessing/process.py", line 314, in _bootstrap
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m     self.run()
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m   File "/usr/local/python3.11.13/lib/python3.11/multiprocessing/process.py", line 108, in run
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m     self._target(*self._args, **self._kwargs)
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m   File "/vllm-workspace/vllm-ascend/vllm_ascend/patch/platform/patch_core.py", line 62, in run_engine_core
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m     raise e
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m   File "/vllm-workspace/vllm-ascend/vllm_ascend/patch/platform/patch_core.py", line 51, in run_engine_core
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m     engine_core.run_busy_loop()
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m   File "/vllm-workspace/vllm/vllm/v1/engine/core.py", line 728, in run_busy_loop
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m     self._process_engine_step()
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m   File "/vllm-workspace/vllm/vllm/v1/engine/core.py", line 754, in _process_engine_step
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m     outputs, model_executed = self.step_fn()
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m                               ^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m   File "/vllm-workspace/vllm/vllm/v1/engine/core.py", line 284, in step
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m     model_output = self.execute_model_with_error_logging(
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m   File "/vllm-workspace/vllm/vllm/v1/engine/core.py", line 270, in execute_model_with_error_logging
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m     raise err
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m   File "/vllm-workspace/vllm/vllm/v1/engine/core.py", line 261, in execute_model_with_error_logging
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m     return model_fn(scheduler_output)
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m            ^^^^^^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m   File "/vllm-workspace/vllm/vllm/v1/executor/abstract.py", line 103, in execute_model
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m     output = self.collective_rpc("execute_model",
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m   File "/vllm-workspace/vllm/vllm/executor/uniproc_executor.py", line 83, in collective_rpc
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m     return [run_method(self.driver_worker, method, args, kwargs)]
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m   File "/vllm-workspace/vllm/vllm/utils/__init__.py", line 3122, in run_method
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m     return func(*args, **kwargs)
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m            ^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m   File "/vllm-workspace/vllm-ascend/vllm_ascend/worker/worker_v1.py", line 277, in execute_model
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m     output = self.model_runner.execute_model(scheduler_output,
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m   File "/usr/local/python3.11.13/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m     return func(*args, **kwargs)
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m            ^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m   File "/vllm-workspace/vllm-ascend/vllm_ascend/worker/model_runner_v1.py", line 1969, in execute_model
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m     max_query_len) = (self._prepare_inputs(scheduler_output,
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m   File "/vllm-workspace/vllm-ascend/vllm_ascend/worker/model_runner_v1.py", line 1407, in _prepare_inputs
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m     mm_embeds, is_mm_embed = self._gather_mm_embeddings(
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m   File "/vllm-workspace/vllm-ascend/vllm_ascend/worker/model_runner_v1.py", line 1099, in _gather_mm_embeddings
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m     assert encoder_output is not None, \
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m            ^^^^^^^^^^^^^^^^^^^^^^^^^^
[P_0] : [1;36m(EngineCore_DP0 pid=2382114)[0;0m AssertionError: Encoder cache miss for 04aa9eda2f42c6ad6af7907bd40d4a9e4915cff8905f059efe8b0da3951a456e.
[P_0] : ERROR 12-17 12:21:19 [async_llm.py:480] AsyncLLM output_handler failed.
[P_0] : ERROR 12-17 12:21:19 [async_llm.py:480] Traceback (most recent call last):
[P_0] : ERROR 12-17 12:21:19 [async_llm.py:480]   File "/vllm-workspace/vllm/vllm/v1/engine/async_llm.py", line 439, in output_handler
[P_0] : ERROR 12-17 12:21:19 [async_llm.py:480]     outputs = await engine_core.get_output_async()
[P_0] : ERROR 12-17 12:21:19 [async_llm.py:480]               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[P_0] : ERROR 12-17 12:21:19 [async_llm.py:480]   File "/vllm-workspace/vllm/vllm/v1/engine/core_client.py", line 846, in get_output_async
[P_0] : ERROR 12-17 12:21:19 [async_llm.py:480]     raise self._format_exception(outputs) from None
[P_0] : ERROR 12-17 12:21:19 [async_llm.py:480] vllm.v1.engine.exceptions.EngineDeadError: EngineCore encountered an issue. See stack trace (above) for the root cause.
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453] Generation failed for request 18da939f-4f99-4fcc-a081-c5d8a00e84e3
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453] Traceback (most recent call last):
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]   File "/vllm-workspace/llm-service/lm_service/workers/vllm/disagg_worker.py", line 442, in _generate
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]     async for request_output in generator:
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]   File "/vllm-workspace/vllm/vllm/v1/engine/async_llm.py", line 387, in generate
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]     out = q.get_nowait() or await q.get()
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]                             ^^^^^^^^^^^^^
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]   File "/vllm-workspace/vllm/vllm/v1/engine/output_processor.py", line 59, in get
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]     raise output
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]   File "/vllm-workspace/vllm/vllm/v1/engine/async_llm.py", line 439, in output_handler
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]     outputs = await engine_core.get_output_async()
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]   File "/vllm-workspace/vllm/vllm/v1/engine/core_client.py", line 846, in get_output_async
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]     raise self._format_exception(outputs) from None
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453] vllm.v1.engine.exceptions.EngineDeadError: EngineCore encountered an issue. See stack trace (above) for the root cause.
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453] Generation failed for request 276253ed-58e8-46b1-a20a-77ee3d3ca98a
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453] Traceback (most recent call last):
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]   File "/vllm-workspace/llm-service/lm_service/workers/vllm/disagg_worker.py", line 442, in _generate
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]     async for request_output in generator:
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]   File "/vllm-workspace/vllm/vllm/v1/engine/async_llm.py", line 387, in generate
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]     out = q.get_nowait() or await q.get()
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]                             ^^^^^^^^^^^^^
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]   File "/vllm-workspace/vllm/vllm/v1/engine/output_processor.py", line 59, in get
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]     raise output
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]   File "/vllm-workspace/llm-service/lm_service/workers/vllm/disagg_worker.py", line 442, in _generate
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]     async for request_output in generator:
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]   File "/vllm-workspace/vllm/vllm/v1/engine/async_llm.py", line 387, in generate
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]     out = q.get_nowait() or await q.get()
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]                             ^^^^^^^^^^^^^
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]   File "/vllm-workspace/vllm/vllm/v1/engine/output_processor.py", line 59, in get
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]     raise output
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]   File "/vllm-workspace/vllm/vllm/v1/engine/async_llm.py", line 439, in output_handler
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]     outputs = await engine_core.get_output_async()
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]   File "/vllm-workspace/vllm/vllm/v1/engine/core_client.py", line 846, in get_output_async
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453]     raise self._format_exception(outputs) from None
[P_0] : ERROR 12-17 12:21:19 [disagg_worker.py:453] vllm.v1.engine.exceptions.EngineDeadError: EngineCore encountered an issue. See stack trace (above) for the root cause.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

--- a/vllm/distributed/ec_transfer/ec_lookup_buffer/mooncake_store.py
+++ b/vllm/distributed/ec_transfer/ec_lookup_buffer/mooncake_store.py
@@ -17,12 +17,11 @@ from dataclasses import dataclass
 from typing import Any, Optional
 
 import numpy as np
-import regex as re
 import torch
 
 from vllm.config import VllmConfig
 from vllm.distributed.ec_transfer.utils.tensor_memory_pool import (
-    InsufficientMemoryError, TensorMemoryPool)
+    TensorMemoryPool)
 from vllm.logger import init_logger
 
 DEFAULT_GLOBAL_SEGMENT_SIZE = 3355443200  # 3.125 GiB
@@ -175,6 +174,7 @@ class ECMooncakeStore:
 
         # Fast transfer init (Use zero-copy methods of mooncake)
         if self.config.fast_transfer:
+            self.pool_lock = threading.Lock()
             self.tensor_pool = TensorMemoryPool(
                 max_block_size=self.config.fast_transfer_buffer_size)
             self.store.register_buffer(self.tensor_pool.base_address,
@@ -239,7 +239,6 @@ class ECMooncakeStore:
             return [None] * len(keys)
 
         buffer_shapes = []
-        buffer_addrs = []
         buffer_dtypes = []
         sizes = []
         exist_ids = []
@@ -258,11 +257,13 @@ class ECMooncakeStore:
             num_elem = math.prod(buffer_shape)
             buffer_size = num_elem * element_size
 
-            buffer_addr = self._pool_allocate(buffer_size)
-            buffer_addrs.append(buffer_addr)
             buffer_dtypes.append(buffer_dtype)
             sizes.append(buffer_size)
             buffer_shapes.append(buffer_shape)
+        with self.pool_lock:
+            buffer_addrs = [
+                self.tensor_pool.allocate(buffer_size) for buffer_size in sizes
+            ]
 
         # Fill None first and
         # Replace valid keys with corresponding buffers.
@@ -272,6 +273,8 @@ class ECMooncakeStore:
             read_bytes = self.store.batch_get_into(valid_keys, buffer_addrs,
                                                    sizes)
         except Exception as e:
+            with self.pool_lock:
+                self.tensor_pool.batch_free(buffer_addrs)
             logger.error("batch_get_into failed: %s", str(e))
 
         # NOTE: should I delay free buffer
@@ -283,7 +286,8 @@ class ECMooncakeStore:
                 results[id] = self.tensor_pool.load_tensor(
                     addr, dtype, shape, device)
 
-            self.tensor_pool.free(addr)
+        with self.pool_lock:
+            self.tensor_pool.batch_free(buffer_addrs)
 
         return results
 
@@ -367,14 +371,13 @@ class ECMooncakeStore:
         meta_values = []
         buffer_addrs = []
         buffer_sizes = []
+        with self.pool_lock:
+            for key, tensor in zip(keys, tensors):
+                buffer_addr = self.tensor_pool.store_tensor(tensor)
+                buffer_size = tensor.numel() * tensor.element_size()
+                buffer_addrs.append(buffer_addr)
+                buffer_sizes.append(buffer_size)
         for key, tensor in zip(keys, tensors):
-            buffer_addr = self._pool_store_tensor(tensor)
-            self.fifo_pool_queue.append(
-                ECMooncakeTensorPoolMetadata(key, buffer_addr))
-            buffer_size = tensor.numel() * tensor.element_size()
-            buffer_addrs.append(buffer_addr)
-            buffer_sizes.append(buffer_size)
-
             meta = {"shape": list(tensor.shape), "dtype": str(tensor.dtype)}
             meta_str = json.dumps(meta)
             meta_bytes = meta_str.encode("utf-8")
@@ -414,6 +417,10 @@ class ECMooncakeStore:
                 ",".join(keys),
                 str(e),
             )
+        finally:
+            if buffer_addrs:
+                with self.pool_lock:
+                    self.tensor_pool.batch_free(buffer_addrs)
 
     async def _batch_put(self, keys: list[str],
                          tensors: list[torch.Tensor]) -> None:
@@ -454,36 +461,3 @@ class ECMooncakeStore:
                 ",".join(keys),
                 str(e),
             )
-
-    # ==============================
-    # Tensor pool helper functions
-    # ==============================
-
-    def _pool_eviction(self) -> None:
-        evicted_buffer = self.fifo_pool_queue.popleft()
-        self.tensor_pool.free(evicted_buffer.addr)
-        key = re.escape(evicted_buffer.key)
-        meta_key = re.escape(self.metadata_key(evicted_buffer.key))
-        count = self.store.remove_by_regex(f"^(?:{key}|{meta_key})$")
-        if count > 2:
-            logger.error("count of key and meta key exceeds 2")
-
-    def _pool_allocate(self, size: int) -> int:
-        while True:
-            try:
-                return self.tensor_pool.allocate(size)
-            except InsufficientMemoryError:
-                if not self.fifo_pool_queue:
-                    raise
-
-                self._pool_eviction()
-
-    def _pool_store_tensor(self, tensor: torch.Tensor) -> int:
-        while True:
-            try:
-                return self.tensor_pool.store_tensor(tensor)
-            except InsufficientMemoryError:
-                if not self.fifo_pool_queue:
-                    raise
-
-                self._pool_eviction()

--- a/vllm/distributed/ec_transfer/utils/tensor_memory_pool.py
+++ b/vllm/distributed/ec_transfer/utils/tensor_memory_pool.py
@@ -265,6 +265,10 @@ class TensorMemoryPool:
 
         return tensor
 
+    def batch_free(self, addrs: list[int]):
+        for addr in addrs:
+            self.free(addr)
+
     def cleanup(self):
         """Cleans up all memory resources and resets the pool state."""
         self.free_lists.clear()

